eth-contracts/contracts/Governance.sol

```diff
@@ -4,284 +4,350 @@ import "./service/registry/RegistryContract.sol";
+        require(_votingPeriod > 0, "Requires non-zero _votingPeriod");
+        votingPeriod = _votingPeriod;
 
-    /** Update voteMagnitudes accordingly */
-
-    // New voter (Vote enum defaults to 0)
-    if (previousVote == Vote.None) {
-      if (_vote == Vote.Yes) {
-        slashProposals[_proposalId].voteMagnitudeYes += voterStake;
-      } else {
-        slashProposals[_proposalId].voteMagnitudeNo += voterStake;
-      }
-      slashProposals[_proposalId].numVotes += 1;
-    }
-    // Repeat voter
-    else {
-      if (previousVote == Vote.Yes && _vote == Vote.No) {
-        slashProposals[_proposalId].voteMagnitudeYes -= voterStake;
-        slashProposals[_proposalId].voteMagnitudeNo += voterStake;
-      }
-      else if (previousVote == Vote.No && _vote == Vote.Yes) {
-        slashProposals[_proposalId].voteMagnitudeYes += voterStake;
-        slashProposals[_proposalId].voteMagnitudeNo -= voterStake;
-      }
-      // If _vote == previousVote, no changes needed to vote magnitudes.
+        require(_votingQuorum > 0, "Requires non-zero _votingQuorum");
+        votingQuorum = _votingQuorum;
     }
 
-    emit SlashProposalVoteSubmitted(
-      _proposalId,
-      voter,
-      _vote,
-      voterStake,
-      previousVote
-    );
-  }
-
-  function evaluateSlashProposalOutcome(uint256 _proposalId)
-  external returns (Outcome proposalOutcome)
-  {
-    require(_proposalId <= lastProposalId && _proposalId > 0, "Must provide valid non-zero _proposalId");
-
-    // Require msg.sender is active Staker.
-    Staking stakingContract = Staking(registry.getContract(stakingProxyOwnerKey));
-    require(
-      stakingContract.totalStakedForAt(
-        msg.sender, slashProposals[_proposalId].startBlockNumber
-      ) > 0,
-      "Caller must be active staker with non-zero stake."
-    );
-
-    // Require proposal votingPeriod has ended.
-    uint256 startBlockNumber = slashProposals[_proposalId].startBlockNumber;
-    uint256 endBlockNumber = startBlockNumber + votingPeriod;
-    require(
-      block.number > endBlockNumber,
-      "Proposal votingPeriod must end before evaluation."
-    );
-
-    // Calculate outcome
-    Outcome outcome;
-    if (slashProposals[_proposalId].numVotes < votingQuorum) {
-      outcome = Outcome.Invalid;
-    }
-    else if (
-      slashProposals[_proposalId].voteMagnitudeYes >= slashProposals[_proposalId].voteMagnitudeNo
-    ) {
-      outcome = Outcome.Yes;
-
-      // Slash target's stake.
-      stakingContract.slash(
-        slashProposals[_proposalId].slashAmount,
-        slashProposals[_proposalId].target
-      );
-    } else {
-      outcome = Outcome.No;
+    // ========================================= Governance Actions =========================================
+
+    function submitProposal(
+        address _targetContract,
+        uint256 _callValue,
+        string calldata _signature,
+        bytes calldata _callData,
+        string calldata _description
```

Comment:
fair point about description length upper bound potentially causing issues. compound also uses string description and has not set a bound, and given that code is audited it's probably fine.
and description crud is not possible since it's not stored on-chain, only emitted. also it's not necessary